benches/ldbc-graphalytics/main.rs

```diff
@@ -136,38 +147,34 @@ fn main() {
 
         let output_inner = output.clone();
         let root = Circuit::build(move |circuit| {
-            if config.profile && Runtime::worker_index() == 0 {
+            if config.profile && is_leader {
                 attach_profiling(dataset, circuit);
             }
 
             let roots = circuit.region("roots", || {
-                circuit.add_source(Generator::new(move || if Runtime::worker_index() == 0 {
+                circuit.add_source(Generator::new(move || if is_leader {
```

Comment:
This can be much cleaner with the new input handle API.  You can use `add_input` to create a "raw" `InputHandle`, which can be used to push pre-batched data to workers.